### ISSUE TITLE
Proposal: eliminate nv as part of the SimpleGraphs datastructure

### ISSUE THREAD

**Issue by sbromberger** [JuliaGraphs/LightGraphs.jl#548]
Right now we have

```
type SimpleGraph{T<:Integer} <: AbstractSimpleGraph
    vertices::UnitRange{T}
    ne::Int
    fadjlist::Vector{Vector{T}} # [src]: (dst, dst, dst)
end
```

We can simplify by simply redefining `nv` and `vertices` for simplegraphs as follows:

```
type SimpleGraph{T<:Integer} <: AbstractSimpleGraph
    ne::Int
    fadjlist::Vector{Vector{T}} # [src]: (dst, dst, dst)
end

nv(g) = eltype(g)(length(fadj(g)))
vertices(g) = one(eltype(g)):nv(g)
```

This doesn't do much in terms of savings, but it does simplify things a bit.

**Comment by jpfairbanks** [comment]
I think that vertices should be `indicies(fadj(g))` for types with non-one based indexing.

**Comment by sbromberger** [comment]
you mean `indices(fadj(g), 1)`? That makes sense.

**Comment by sbromberger** [comment]
But this will probably break things mysteriously. I wonder where we've assumed vertex indices start at 1.

**Comment by jpfairbanks** [comment]
Well all of our current types start at 1, so it should be fine.
The Julia Manual says that this kind of breakage is expected when you generalize libraries that assume that things start at 1. It says "these errors will help you find places to generalize the code" or something like that.

**Comment by sbromberger** [comment]
It'll be fine for our current code, but do we really want to relax our design to allow vertex indices to start at other than 1?

**Comment by sbromberger** [comment]
We can't use `indices` in any case, because there's no way to change the type of the `Base.OneTo` that it creates. It's always an `Int64`, and we assume that both `nv` and `vertices` will yield integers of the same type as the graph vertex indices.

**Comment by sbromberger** [comment]
Leaving this open for further discussion, but I just merged the original code into #541. I found a couple places where we were relying on the internal data structures of Graph and DiGraph, so I fixed them up.

**Comment by sbromberger** [comment]
Closing this out since it works / passes all tests.
